db-service/lib/SQLService.js

```diff
@@ -114,6 +114,11 @@ class SQLService extends DatabaseService {
    * @type {Handler}
    */
   async onSELECT({ query, data }) {
+    if (query.SELECT.from?.join && !query.SELECT.columns) {
```

Comment:
What is the `join` is not on root level ?

```cql
SELECT ID FROM (
  SELECT * FROM
  Books
  LEFT JOIN
  Authors
  ON
  Books.author_ID = Authors.ID
)
```

Maybe it makes sense to move it into `cqn2sql` ([code](https://github.com/cap-js/cds-dbs/blob/main/db-service/lib/cqn2sql.js#L208)). Which will reject it on all levels.

Reply:
I'll double check